internal/app/admin/http/cache.go

```diff
@@ -151,31 +151,25 @@ func cacheDumpHandler(o *orchestrator.Orchestrator) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
 		cacheKey := getParam(req.URL.Path)
 		c := orchestrator.Orchestrator.GetReadOnlyCache(*o)
-		var keysToPrint []string
-
-		// If wildcard suffix provided, output all cache entries that match given prefix.
-		// If no key is provided, output the entire cache.
-		if hasWildcardSuffix(cacheKey) {
-			// Retrieve all keys
-			allKeys, err := orchestrator.Orchestrator.GetDownstreamAggregatedKeys(*o)
-			if err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
-				fmt.Fprintf(w, "error in getting cache keys: %s", err.Error())
-				return
-			}
+		keysToPrint, err := getRelevantKeys(o, cacheKey, w)
+		if err == nil {
+			printCacheEntries(keysToPrint, c, w)
+		}
+	}
+}
 
-			// Find keys that match prefix of wildcard
-			rootCacheKeyName := strings.TrimSuffix(cacheKey, "*")
-			for potentialMatchKey := range allKeys {
-				if strings.HasPrefix(potentialMatchKey, rootCacheKeyName) {
-					keysToPrint = append(keysToPrint, potentialMatchKey)
-				}
+func clearCacheHandler(o *orchestrator.Orchestrator) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		if req.Method == http.MethodPost {
+			cacheKey := getParam(req.URL.Path)
+			keysToClear, err := getRelevantKeys(o, cacheKey, w)
+			if err == nil {
+				clearCacheEntries(keysToClear, o, w)
 			}
-		} else {
-			// Otherwise return the cache entry corresponding to the given key.
-			keysToPrint = []string{cacheKey}
+			return
 		}
-		printCacheEntries(keysToPrint, c, w)
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		_, _ = w.Write([]byte("Only POST is supported\n"))
```

Comment:
i think we dont need this error message. `StatusMethodNotAllowed` is self explanatory